### PR TITLE
Crazy callback wrappers

### PR DESCRIPTION
**Pull request by Malacath-92**
- I may have gone overboard with the safe callbacks because I wanted to replace all our manually made callbacks and we weren't super consistent (hence the front and back binding). Most use-cases are gonna be very simple however, something like the socket code
- I also bumped to clang 15.0.2 while I was at it
- made `handle_function` auto-cast entities so we don't have to manually `lua["cast_entity"](ent)` all the time

Still have to test, hopefully sometime today.

### CODE REVIEW

**Comment by Malacath-92** [comment]
@Dregu for your async http stuff you can use this simply as `make_safe_cb<void(std::string_view)>(std::move(lua_cb))` to get a thread-safe callback.

File: src/game_api/script/usertypes/vtables_lua.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        EntityVTable,
+        VTableEntry<"damage", 0x30, void(Entity*, int8_t, uint32_t, float*, float*, uint16_t, uint8_t)>>;
+    static MovableVTable movable_vtable(lua, lua["Movable"], "ENTITY_OVERRIDE");
+
+    using ContainerVTable = HookableVTable<
+        Entity,
+        CallbackType::Entity,
+        MovableVTable,
+        VTableEntry<"open", 0x18, void(Movable*)>>;
+    static ContainerVTable container_vtable(lua, lua["Container"], "ENTITY_OVERRIDE");
```
[the remark this answers is not in the document]
Comment (Dregu) [reply src/game_api/script/usertypes/vtables_lua.cpp:38]: Ok I'm confused now. Why does `set_on_open(shotgun_uid, ...)` (on main) only trigger if holding up while shooting?

[the remark this answers is not in the document]
Comment (Mr-Auto) [reply src/game_api/script/usertypes/vtables_lua.cpp:38]: hmm.. maybe missing parameter? :thinking:

[the remark this answers is not in the document]
Comment (Mr-Auto) [reply src/game_api/script/usertypes/vtables_lua.cpp:38]: just checked with plugin and the function gets called every time you shot, not only when holding up

[the remark this answers is not in the document]
Comment (Dregu) [reply src/game_api/script/usertypes/vtables_lua.cpp:38]: Well the callback does not, but moving this to EntityVTable works always 🤷🏻‍♂️

[the remark this answers is not in the document]
Comment (Mr-Auto) [reply src/game_api/script/usertypes/vtables_lua.cpp:38]: On main there is code `opener->movey > 0` i guess, the idea was to only call the callback when you open container, not when you damage it or whatever

[the remark this answers is not in the document]
Comment (Dregu) [reply src/game_api/script/usertypes/vtables_lua.cpp:38]: Well good thing we're getting rid of that then

[the remark this answers is not in the document]
Comment (Dregu) [reply src/game_api/script/usertypes/vtables_lua.cpp]: Yep I guess the deprecated set_on_open should stay as it was, even though the old behavior was weird in the first place.

**Pull request merged**